### PR TITLE
[feat] Add Waiting Room

### PR DESCRIPTION
**Pull request by TejusR**

### CODE REVIEW

**Comment by NavenAllen** [comment]
Have you made the changes that @siliconcupcake suggested?

**Comment by NavenAllen** [comment]
Also, there is no way to know the game code when you have created the game and are waiting. Add a game code field showing the code and something like click to copy so it is easier to share it.

File: src/game-engine/components/HomePage/HomePage.js [hunk 1]
```diff
@@ -282,6 +282,7 @@ const HomePage = (props) => {
 											},
 											inputMode: 'text'
 										}}
+										onChange={handleNameInputChange}
```
Comment (siliconcupcake) [inline src/game-engine/components/HomePage/HomePage.js]: This is repetitive ?

File: src/game-engine/components/WaitingRoom/WaitingRoom.js [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+												variant="contained"
+												color="primary"
+												disabled={
+													props.players.length < 6
+												}
+												onClick={props.startGame}
+											>
+												Start Game
+											</Button>
+										) : null}
```
Comment (siliconcupcake) [inline src/game-engine/components/WaitingRoom/WaitingRoom.js]: For players who aren't owner, add button to leave game.

File: src/game-engine/components/GamePage/GamePage.js [hunk 3]
```diff
@@ -4,41 +4,57 @@ import { Engine } from '../../engine'
...
-					left: '50vw',
-					top: '60vh'
-				}}
-				onClick={() => deal(4)}
-			>
-				Deal
-			</button>
-			{game}
-			{props.cardSelected !== undefined ? <AskCard /> : null}
+			{props.gameData.isActive != true ? (
```
Comment (siliconcupcake) [inline src/game-engine/components/GamePage/GamePage.js:14]: Move displaying game code into the waiting room, add it to title there and also call game actions directly from within the waiting room.

**Review by siliconcupcake — CHANGES_REQUESTED** [review]

**Review by siliconcupcake — COMMENTED** [review]
Also, there are a lot of unnecessary comments on the WaitingRoom component. Clear that up too ?

File: src/game-engine/components/HomePage/HomePage.js (hunk 1 above)
Comment (TejusR) [reply src/game-engine/components/HomePage/HomePage.js]: Removed it.

**Review by siliconcupcake — DISMISSED** [review]

**Review by siliconcupcake — APPROVED** [review]

**Pull request merged**